bookdown/08-interpretation.Rmd

```diff
@@ -1,10 +1,165 @@
+
+
+
+**Predicting**
+  
+It is also interesting to see how well a model has trained a data set. For this section, exactly as was recommended in section [2.4](https://mlr3book.mlr-org.com/train-predict.html) of the mlr3book, 80% of the penguin data set will be used for the training set and 20% for the test set:
+  
+```{r pred, message=FALSE, warning=FALSE}
+train_set = sample(task_peng$nrow, 0.8 * task_peng$nrow)
+test_set = setdiff(seq_len(task_peng$nrow), train_set)
+learner$train(task_peng, row_ids = train_set)
+prediction = learner$predict(task_peng, row_ids = test_set)
+```
+
+And the plots below show the results for `FeatureImp` and `FeatureEffect`.
+
+`FeatureImp`:
+```{r featimppred, message=FALSE, warning=FALSE, fig.cap='FeatImp train data set',  fig.align='center'}
+model = Predictor$new(learner, data = penguins[train_set,], y = "species")
+effect = FeatureImp$new(model, loss ="ce" )
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+The results of the train set for `FeatureImp` (and `FeatureEffect`) are similar to the results above, which is expected. 
+
+```{r featimptest, message=FALSE, warning=FALSE, fig.cap='FeatImp test data set', fig.align='center'}
+model = Predictor$new(learner, data = penguins[test_set,], y = "species")
+effect = FeatureImp$new(model, loss ="ce" )
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+
+Depending on which batch of input is sampled into the train data set, the reader will either see under- or overfitting in the test set.
+
+
+`FeatureEffects`:
+```{r feateffreal, message=FALSE, warning=FALSE, fig.cap='FeatEffect train data set', fig.align='center'}
+model = Predictor$new(learner, data = penguins[train_set,], y = "species")
+effect = FeatureEffects$new(model)
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+
+```{r featefftest, message=FALSE, warning=FALSE, fig.cap='FeatEffect test data set',  fig.align='center'}
+model = Predictor$new(learner, data = penguins[test_set,], y = "species")
+effect = FeatureEffects$new(model)
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+
+As with the case with `FeatureImp` the results here from the test set also show under-/ overfitting problems. This would be a good opportunity for the reader to attempt to resolve the under- or overfitting by playing with the amount of features and the amount of data used for both the test and train data sets. Be sure to not change the line `train_set = sample(task_peng$nrow, 0.8 * task_peng$nrow)` as it will randomly sample the data again.
```

Comment:
Please repeat the example from above here also with `FeatureImp`.
Also this paragraph is not correct, since it will not necessarily show under or overfitting of the model. Rather it might show that we get an over-optimistic Feature Importance result.

Reply:
What do you mean repeat the example?